AssociativeTable.cs

```diff
@@ -30,22 +26,20 @@ private void DoubleArraySize()
     /// </summary>
     /// <param name="tempKey">The key associated with the Tuple.</param>
     /// <param name="tempValue">The value associated with the Tuple.</param>
-    public void Put(TKey tempKey, TValue tempValue)
+    public override void Put(TKey tempKey, TValue tempValue)
```

Comment:
lets change the name of parameters `tempKey` and `tempValue` to just `key` and `value`, remember to update the docstring

Reply:
we can optimize it by adding an instance's attribute that holds the index of the first empty node, we can make it O(1) instead of O(n)